src/main/java/net/dries007/tfc/ConfigTFC.java

```diff
@@ -46,614 +31,1423 @@ public static void onConfigChangedEvent(ConfigChangedEvent.OnConfigChangedEvent
+            public double collapseChance = 0.1;
+
+            @Config.Comment("Chance that collapsing blocks propagate the collapse. Influenced by distance from epicenter of collapse.")
+            @Config.RangeDouble(min = 0, max = 1)
+            @Config.LangKey("config." + MOD_ID + ".general.fallable.propagateCollapseChance")
+            public double propagateCollapseChance = 0.55;
+
+            @Config.Comment("Should chiseling raw stone blocks cause collapses?")
+            @Config.LangKey("config." + MOD_ID + ".general.fallable.chiselCausesCollapse")
+            public boolean chiselCausesCollapse = true;
+        }
+
+        public static final class DifficultyCFG
+        {
+            @Config.Comment("This controls how many animals (any kind) are loaded (spawned) / player. Higher values means more animals near a player, which in turn raises difficulty and meat abundance (caution, a value too high can also negatively impact performance).")
+            @Config.RangeInt(min = 0)
+            @Config.LangKey("config." + MOD_ID + ".general.difficulty.animalSpawnCount")
+            public int animalSpawnCount = 75;
+
+            @Config.Comment("This controls how many mobs are loaded (spawned) / player. Higher values means more mobs near a player, which in turn raises difficulty (caution, a value too high can also negatively impact performance)")
+            @Config.RangeInt(min = 0)
+            @Config.LangKey("config." + MOD_ID + ".general.difficulty.mobSpawnCount")
+            public int mobSpawnCount = 140;
+
+            @Config.Comment("Prevent mob spawning on surface?")
+            @Config.LangKey("config." + MOD_ID + ".general.difficulty.preventMobsOnSurface")
+            public boolean preventMobsOnSurface = true;
+
+            @Config.Comment("Give wrought iron weapons to vanilla mobs?")
+            @Config.LangKey("config." + MOD_ID + ".general.difficulty.giveVanillaMobsEquipment")
+            public boolean giveVanillaMobsEquipment = true;
+        }
+
+        public static final class TreeCFG
+        {
+            @Config.Comment({"Enable trees being fully cut by axes.",
+                "Disable it if you want other mods to handle tree felling."})
+            @Config.LangKey("config." + MOD_ID + ".general.tree.enableFelling")
+            public boolean enableFelling = true;
+
+            @Config.Comment("Should logs require tools (axes and saws, or hammers for sticks) to be cut? If false, breaking logs with an empty hand will not drop logs.")
+            @Config.LangKey("config." + MOD_ID + ".general.tree.requiresAxe")
+            public boolean requiresAxe = true;
+
+            @Config.Comment("If false, leaves will not drop saplings.")
+            @Config.LangKey("config." + MOD_ID + ".general.tree.enableSaplings")
+            public boolean enableSaplings = true;
+
+            @Config.Comment("Log return rate of stone axes (eg: How efficiently it is)")
```

Comment:
efficient not efficiently
```suggestion
            @Config.Comment("Chance per log for an item to drop when using a stone axe.")
```